### ISSUE TITLE
logger not firing

### ISSUE THREAD

**Issue by Tdarnell** [robertwayne/cogwatch#28]
I am not sure if this is unique to the disnake library, but the logger.info calls in the else after the try blocks are not being called.

In fact any code added after the `await self.client.reload_extension(cog_dir)` never gets run ... 

I can confirm from log statements in my cog's setup and taredown functions that the extension is being reloaded, it is just strange that nothing after the await runs. 

```python
    async def reload(self, cog_dir: str):
        """Attempts to atomically reload the file into the client."""
        try:
            logger.info(f'{self.CBOLD}{self.CGREEN}[Cog Reloaded]{self.CEND} Before') # This one will print correctly
            await self.client.reload_extension(cog_dir)
            logger.info(f'{self.CBOLD}{self.CGREEN}[Cog Reloaded]{self.CEND} After') # This line will never be run
        except commands.NoEntryPointError:
            logger.info(
                f'{self.CBOLD}{self.CRED}[Error]{self.CEND} Failed to reload {self.CBOLD}{cog_dir}{self.CEND}; no entry point found.'
            )
        except commands.ExtensionNotLoaded:
            logger.info(f'Cannot reload {cog_dir} because it is not loaded.')
        except Exception as exc:
            self.cog_error(exc)
        else:
            logger.info(f'{self.CBOLD}{self.CGREEN}[Cog Reloaded]{self.CEND} {cog_dir}') # This line will never be run
```

**Comment by robertwayne** [comment]
I had a similar issue when I initially added support for `nextcord`, but I thought I had fixed it. To be honest, I'm not even sure why it wouldn't fire. There's nothing blocking it.

I'll take a deeper look and see if I can figure out the root cause.

**Comment by robertwayne** [comment]
This should be fixed in v3.2.0 - I tested it on every supported library, but if you run into any issues, let me know.
